Library/String+CurrencyTests.swift

```diff
@@ -0,0 +1,409 @@
+import XCTest
+@testable import Library
+
+// swiftlint:disable line_length
+final class StringCurrencyTests: XCTestCase {
+  func testCurrencySymbol_NoSymbol() {
```

Comment:
The first batch of tests is testing only limited inputs (i.e. currency, amount, fractions, etc.).

The second batch of tests (prefixed `testCombined..`) is testing some common combinations.